src/OpenTelemetry/Resources/OtelEnvResourceDetector.cs

```diff
@@ -41,7 +41,7 @@ public Resource Detect()
             }
             catch (Exception ex)
             {
-                OpenTelemetrySdkEventSource.Log.ResourceDetectorFailed("OtelEnvResourceDetector", ex.Message);
+                OpenTelemetrySdkEventSource.Log.ResourceDetectorFailed(nameof(OtelEnvResourceDetector), ex.Message);
```

Comment:
Should we switch this catch to `SecurityException` to match the others?

Reply:
I am not sure if this is the only exception type that can happen here.

Reply:
I think it OK to just catch `SecurityException` but defer to @cijothomas. I'm basically just echoing stuff that was discussed on this PR: https://github.com/open-telemetry/opentelemetry-dotnet/pull/2123#discussion_r667134036

Reply:
I tested it as much as possible and I think it is OK to just catch `SecurityException`